client/my-sites/email/inbox/index.js

```diff
@@ -33,26 +36,38 @@ const InboxManagement = ( { domains } ) => {
 		);
 	}
 
-	const domainsWithSubscriptions = domains.filter(
-		( domain ) => ! domain.isWPCOMDomain && hasPaidEmailWithUs( domain )
+	const nonStagingDomains = domains.filter( ( domain ) => ! domain.isWPCOMDomain );
+
+	// Find out if the site has at least one mailbox in the site
+	const hasAnyMailboxes = domains.some(
+		( domain ) =>
+			( nonStagingDomains &&
+				// it may be worth also checking for has(EmailProduct)WithUs( domain ) for each branch below
+				getGSuiteMailboxCount( domain ) > 0 ) ||
+			getConfiguredTitanMailboxCount( domain ) > 0
 	);
```

Comment:
I think we can now use the `nonStagingDomains` as base.

```suggestion
	const hasAnyMailboxes = nonStagingDomains.some(
		( domain ) =>
			( getGSuiteMailboxCount( domain ) > 0 ) ||
			getConfiguredTitanMailboxCount( domain ) > 0
	);
```